File: spoj/DOTAA.cpp
#include <iostream>
using namespace std;

int main(){
	int T,m,n,D,i,j,k,l;
	cin >> T;
	for(i=0;i<T;i++){
		cin >> n >> m >> D;
		j = 0;
		for(l=0;l<n;l++){
			cin >> k;
			if(1.0*k/D > (int)(k/D)) {
				j += k/D;
			} else if(1.0*k/D == (int)k/D) {
				j += k/D - 1;
			}
		}
		if (j>=m)
			cout<<"YES"<<endl;
		else
			cout<<"NO"<<endl;
		}
	}

File: spoj/DOTAA.py
import math
for _ in range(input()):
	n,m,D = map(int,raw_input().split())
	for i in range(n):
		a = input()
		a = math.ceil(a/D)
		m -= a-1
	if(m<=0):
		print "YES"
	else:
		print "NO"

